src/D2L.CodeStyle.Analyzers/Immutability/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -110,5 +119,12 @@ INamedTypeSymbol analyzedType
 
 			return false;
 		}
+
+		private static bool IsFromGeneratedResourcesFile( ISymbol symbol ) {
+			var tree = symbol.DeclaringSyntaxReferences.Single().SyntaxTree;

```

Comment:
C#9 adds support for partial methods (separating declaration and definition). Don't recall if that applied to fields and members as well.

Reply:
Doesn't look like it.